core/src/main/groovy/noe/tomcat/configure/SecureHttpConnectorTomcat.groovy

```diff
@@ -25,12 +29,26 @@ public class SecureHttpConnectorTomcat extends ConnectorTomcatAbstract<SecureHtt
   private String sslProtocol
   private String keystoreFile
   private String keystorePass
+  private String keystoreType
   private boolean clientAuth
+  private String truststoreFile
+  private String truststorePass
+  private String truststoreType
+  private String sslImplementationName
+  private boolean sslEnabledProtocols
 
   // SSL APR
+  private String sslCACertificateFile
   private String sslCertificateFile
   private String sslCertificateKeyFile
   private String sslPassword
+
+  static Platform platform = new Platform() /// platform identification
```

Comment:
Can we have all these default fields in proper methods to have code more semantic (nothing indicates that those values are used as defaults and it can cause unnecessary headache the next code maintainer to understand what is going on.

So you can have in a body smth like: `setKeystoreFile(new File(getDefultSslDir(), "server.jks"))`
and you ucan have:
````
public File getDefultSslDir() {
  File sslDir = new File (new Platform().tmpDir, "ssl")
  return new File(sslDir, "self_signed")
}
````
All code is just for illustration from the top of my head - this can be improved in any way.

Reply:
And BTW static is evil what should be avoided as much as possible (if you really don't have good case).

more on this for instane here
https://dzone.com/articles/why-static-bad-and-how-avoid